### PR TITLE
Update wdio/cli to also build for CJS

### LINKED ISSUE #9869: [🐛 Bug]: CJS require of @wdio/cli and @wdio/reporter not implemented
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.4

### Node.js Version

16.13.2

### Mode

WDIO Testrunner

### Which capabilities are you using?

_No response_

### What happened?

Tests are written in a cjs project with the testrunner started programmatically. When upgrading to v8 some wdio modules are no longer possible to require. 
```
const Launcher = require('@wdio/cli').default
const WDIOReporter = require('@wdio/reporter').default;
```
fails since they only support esm import.
A temporary workaround is to use ```const { Launcher } = await import('@wdio/cli');```
but in the case of @wdio/reporter that is a bit trickier.

### What is your expected behavior?

To be able to use 
```
const Launcher = require('@wdio/cli').default
const WDIOReporter = require('@wdio/reporter').default;
```

### How to reproduce the bug.

Use 
```
require('@wdio/cli').default
```
or
```
require('@wdio/reporter').default
```
in v8

### Relevant log output

```typescript
Launcher failed to start the test Error [ERR_REQUIRE_ESM]: require() of ES Module
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### LINKED ISSUE #10119: [🐛 Bug]: @wdio/cli Launcher not compatible to CJS context
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

v8

### Node.js Version

v18.12.0

### Mode

Standalone Mode

### Which capabilities are you using?

_No response_

### What happened?

When upgrading from v7 to v8 the @wdio/cli Launcher is not compatible with CJS context projects anymore.
Looks like the Launcher only works with the ESM context.

 >  NX   require() of ES Module /node_modules/@wdio/cli/build/index.js from /tools/builders/src/e2e/index.js not supported.
 
 While it seems simple to just convert your project from CJS to ESM, it's not that easy with a massive monorepo where these changes take a bit of time. Would be great to be able to upgrade to v8 without having to change our setup to ESM.

### What is your expected behavior?

Expect to be able to use @wdio/cli Launcher with either ESM or CJS projects.

### How to reproduce the bug.

Create a CJS script to start Webdriverio tests with the @wdio/cli Launcher.

### Relevant log output

```typescript
>  NX   require() of ES Module /node_modules/@wdio/cli/build/index.js from /tools/builders/src/e2e/index.js not supported.

   Instead change the require of /node_modules/@wdio/cli/build/index.js in /tools/builders/src/e2e/index.js to a dynamic import() which is available in all CommonJS modules.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by DirkoOdendaal**
## Proposed changes

 Fixes #9869

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

Enable wdio/cli to also build for CJS.

### CODE REVIEW

File: packages/wdio-cli/src/utils.ts [hunk 1]
```diff
@@ -376,11 +372,11 @@ export async function detectCompiler(answers: Questionnair) {
...
 /**
  * Check if package is installed
- * @param {string} package to check existance for
+ * @param {string} package to check existence for
  */
-export async function hasPackage(pkg: string) {
+export function hasPackage(pkg: string) {
     try {
-        await resolve(pkg, import.meta.url)
+        require.resolve(pkg)
```
Comment (christian-bromann) [inline packages/wdio-cli/src/utils.ts:379]: Why this change? `require` doesn't work in ESM

Comment (DirkoOdendaal) [reply packages/wdio-cli/src/utils.ts:379]: Because `import.meta.url` does not work in CJS.
